src/scitokens_internal.cpp

```diff
@@ -312,6 +312,9 @@ void
 SciToken::deserialize(const std::string &data, const std::vector<std::string> allowed_issuers) {
     m_decoded.reset(new jwt::decoded_jwt(data));
 
+    // Set all the claims
+    m_claims = m_decoded->get_payload_claims();
```

Comment:
Can you move this to the end of the function?

That way, we only make a copy of the claims if the validator passes.